src/raftstore/store/engine.rs

```diff
@@ -240,11 +239,63 @@ pub trait Mutable: Writable {
 impl Mutable for DB {}
 impl Mutable for WriteBatch {}
 
+const MAX_DELETE_KEYS_COUNT: usize = 10000;
+
+/// `delete_all_in_range` fast deletes data of all cfs in range [`start_key`, `end_key`).
+/// It uses rocksdb `delete_file_in_range` first, then scans the left keys and
+/// uses `WriteBatch` to deletes them.
+/// Note: this function is dangerous and not guarantees consistence. If `delete_file_in_range`
+/// finishes successfully but commit following `WriteBatch` failed, some keys are really deleted
+/// and can't be recovered.
+pub fn delete_all_in_range(db: &DB, start_key: &[u8], end_key: &[u8]) -> Result<()> {
+    for cf in db.cf_names() {
+        try!(delete_in_range_cf(db, cf, start_key, end_key));
+    }
+
+    Ok(())
+}
+
+pub fn delete_in_range_cf(db: &DB, cf: &str, start_key: &[u8], end_key: &[u8]) -> Result<()> {
+    let handle = try!(rocksdb::get_cf_handle(db, cf));
+    try!(db.delete_file_in_range_cf(*handle, start_key, end_key));
```

Comment:
If we use this api to destroy region, we'd better increase region size, this api can benefit a lot just when the deleted chunk is large.